sycl/doc/extensions/proposed/sycl_ext_oneapi_complex_marray.asciidoc

```diff
@@ -0,0 +1,379 @@
+|`mgenfloat get_real(const mgencomplex& x, std::integer_sequence int_seq);`
+|Returns a sequence of real components of the complex number x. Each element should be constructed from the corresponding index within `int_seq` and the returned marray size should be the same as the `int_seq` size.
+|`mgenfloat get_imag(const mgencomplex& x, std::integer_sequence int_seq);`
+|Returns a sequence of imaginary components of the complex number x. Each element should be constructed from the corresponding index within `int_seq` and the returned marray size should be the same as the `int_seq` size.
+|`void set_real(mgencomplex& x, const mgenfloat& y);`
+|Set each element of the real components in x to the corresponding element in y.
+|`void set_imag(mgencomplex& x, const mgenfloat& y);`
+|Set each element of the imaginary components in x to the corresponding element in y.
+|`void set_real(mgencomplex& x, const genfloat y);`
+|Set each element of the real components in x to the decimal number y.
+|`void set_imag(mgencomplex& x, const genfloat y);`
+|Set each element of the imaginary components in x to the decimal number y.
+|===
+
+=== Mathematical operations
+
+This proposal extends `sycl::ext::oneapi` namespace math functions to accept
+`mgencomplex` for the SYCL math functions, `abs`, `acos`, `asin`, `atan`,
+`acosh`, `asinh`, `atanh`, `arg`, `conj`, `cos`, `cosh`, `exp`, `log`, `log10`,
+`norm`, `polar`, `pow`, `proj`, `sin`, `sinh`, `sqrt`, `tan`, and `tanh`.
+For math functions with two parameters marray-scalar and scalar-marray overloads
+are added.
+
+The functions execute as-if the math operation is performed elementwise across the
+marray. The math function between each element should follow the C++ 
+standard for handling NaN's and Inf values. 
+
+The proposal additionally adds overloads between marrays and scalar inputs.
+Overloads with marray's and scalar parameters should execute the operation 
+across the marray while keeping the scalar value constant.
+
+```C++
+namespace sycl {
+namespace ext {
+namespace oneapi {
+
+    mgenfloat abs(const mgencomplex& x);
+
+    mgencomplex acos(const mgencomplex& x);
+
+    mgencomplex asin(const mgencomplex& x);
+
+    mgencomplex atan(const mgencomplex& x);
+
+    mgencomplex acosh(const mgencomplex& x);
+
+    mgencomplex asinh(const mgencomplex& x);
+
+    mgencomplex atanh(const mgencomplex& x);
+
+    mgenfloat arg(const mgencomplex& x);
+
+    mgencomplex conj(const mgencomplex& x);
+
+    mgencomplex cos(const mgencomplex& x);
+
+    mgencomplex cosh(const mgencomplex& x);
+
+    mgencomplex exp(const mgencomplex& x);
+
+    mgencomplex log(const mgencomplex& x);
+
+    mgencomplex log10(const mgencomplex& x);
+
+    mgenfloat norm(const mgencomplex& x);
+
+    mgencomplex polar(const mgenfloat& rho, const mgenfloat& theta);
+    mgencomplex polar(const mgenfloat& rho, const genfloat& theta = 0);
+    mgencomplex polar(const genfloat& rho, const mgenfloat& theta);
```

Comment:
```suggestion
    mgencomplex polar(const mgenfloat& rho, genfloat theta = 0);
    mgencomplex polar(genfloat rho, const mgenfloat& theta);
```

Should these (and other) mathematical operations be changed to pass-by-value?

Reply:
Yes pow and polar have these overloads changed to pass-by-value